scipy/signal/spectral.py

```diff
@@ -1417,15 +1419,19 @@ def _spectral_helper(x, y, fs=1.0, window='hann', nperseg=None, noverlap=None,
     # Padding occurs after boundary extension, so that the extended signal ends
     # in zeros, instead of introducing an impulse at the end.
     # I.e. if x = [..., 3, 2]
-    # even_ext then pad -> [..., 3, 2, 2, 3, 0, 0, 0]
-    # pad then even_ext -> [..., 3, 2, 0, 0, 0, 2, 3]
+    # extend then pad -> [..., 3, 2, 2, 3, 0, 0, 0]
+    # pad then extend -> [..., 3, 2, 0, 0, 0, 2, 3]
+
+    boundary_funcs = {'even': even_ext,
+                      'odd': odd_ext,
+                      'constant': const_ext,
+                      'zeros': zero_ext}
 
-    if boundary == 'even':
-        # Even extension means first window doesn't get distorted by a
-        # shift in the DC signal level as much when x[0] != 0
-        x = even_ext(x, nperseg//2, axis=-1)
+    if boundary is not None:
+        ext_func = boundary_funcs[boundary]
```

Comment:
For a cleaner error it might be worth checking `if boundary not in boundary_funcs: raise ValueError(...)`